adding `spring security sample` folder to this repo

ramya25 opened the pull request.

Commit:
adding `spring security sample` folder to this repo

Commit:
updating README

Commit:
updating README

sangonzal commented on spring-security-web-app/README.md:
```diff
@@ -0,0 +1,225 @@
+
+#### Running from IDE
+
+If you running you web application from an IDE, click on **run**, then navigate to the home page of the project. For this sample, the standard home page URL is <http://localhost:8080>
```
If you are running the web application*

ramya25 replied:
updated

navyasric commented on spring-security-web-app/README.md:
```diff
@@ -0,0 +1,225 @@
+
+To register the project, you can follow the steps in the paragraphs below:
+
+#### First step: choose the Azure AD tenant where you want to create your applications
```
This header is repeated on line 71. We can delete this line.

ramya25 replied:
yeah sure

navyasric commented on spring-security-web-app/README.md:
```diff
@@ -0,0 +1,225 @@
+
+The relevant code for this sample is in the `AppConfiguration.java` file.
+
+This class extends `WebSecurityConfigurerAdapter` from which the `configure` method is overridden.
```
Can we add a line about what the overridden method is doing in relation to authentication?

ramya25 replied:
added

navyasric commented on spring-security-web-app/README.md:
````diff
@@ -0,0 +1,225 @@
+This class extends `WebSecurityConfigurerAdapter` from which the `configure` method is overridden.
+
+```Java
+public void configure(HttpSecurity http) throws Exception {
````
Can we also include the "@EnableOAuth2Sso" etc on top of the method since these are relevant to the auth working...

ramya25 replied:
Yes, included.

navyasric commented on spring-security-web-app/README.md:
```diff
@@ -0,0 +1,225 @@
+1. The Java web application obtains an:
+
+   - Id Token from Azure Active Directory (Azure AD) to sign in an user
+   - Access token that is validated using Spring security.
```
This sample does not appear to be doing any token validation as it is only a web app signing in a user? I do not see any token validation code.

ramya25 replied:
token validation would happen in this class `WebSecurityConfigurerAdapter`, if we add more conditions in configure method (like `.access("#oauth2.hasScope('" + accessAsUserScope + "')");` in our webapi sample)

navyasric replied:
@ramya25 Yes I understand that it would happen by adding those lines in a webapi sample. But this particular sample is not doing it so we should remove the line about AT is validated from the Readme as it is misleading to developers. Santiago also confirms below there is only logging in a user in this sample.

ramya25 replied:
sure will remove it.

navyasric commented on spring-security-web-app/README.md:
```diff
@@ -0,0 +1,225 @@
+
+### Overview
+
+This sample demonstrates a Java web application showcasing how to use spring security to validate the access token obtained using OAuth2.0 and authenticates the user.
```
I do not see any token validation happening in this sample. I think this sample demonstrates signing in or authenticating a user with Azure AD using Spring Security and OAuth 2.0.

sangonzal replied:
No token validation, just using Spring Security for logging in an user via Oauth2

ramya25 replied:
updated

navyasric reviewed (changes requested):
Looks good. Added a few suggestions.

Can we also edit the top level Readme in this repo with a link to this folder saying:
"For a java web app sample using Spring Security framework take a look at <link to folder>"

Thanks.

Commit:
Implementing PR comments and updating README

Commit:
Updating README

ramya25 commented:
> Looks good. Added a few suggestions.
> 
> Can we also edit the top level Readme in this repo with a link to this folder saying:
> "For a java web app sample using Spring Security framework take a look at "
> 
> Thanks.

Updated top level README with link to Spring Security sample link
